lib/src/lifecycle_module.dart

```diff
@@ -889,6 +889,17 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
     }
   }
 
+  @mustCallSuper
+  @override
+  @protected
+  Future<Null> onDispose() async {
+    if (isInstantiated || isUnloaded) {
+      return;
+    }
+
+    _childModules.clear();
```

Comment:
We should add a test for this. I think the easiest way to do that would be to insert a `tearDown` step right here:
https://github.com/Workiva/w_module/blob/master/test/lifecycle_module_test.dart#L1502

```dart
tearDown(() {
  if (withChild) {
    expect(module.childModules, isEmpty);
  }
});
```

Reply:
I would have expected to see this right after line 888 where we are disposing of the child modules. It seems like you guys have found a bug that is preventing us from hitting line 888 where we dispose of the child modules--is anyone trying to figure out why that's not getting hit?